### PR TITLE
[Tensorflow] Fix import

### PR DESCRIPTION
**Pull request by gtopper**
[ML-5916](https://iguazio.atlassian.net/browse/ML-5916)

tensorflow 2.16.1 breaks `tensorflow.keras.callbacks`.

`tensorflow.python.keras.callbacks` works with tensorflow 2.16.1 as well as with previous versions.

[ML-5916]: https://iguazio.atlassian.net/browse/ML-5916?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by guy1992l — CHANGES_REQUESTED** [review]
Good catch Gal. For backward compatibility please check tf.__version__ and import accordingly

**Comment by gtopper** [comment]
@guy1992l, from my testing, `import tensorflow.python.keras.callbacks` works with older versions as well.

File: mlrun/frameworks/tf_keras/callbacks/logging_callback.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import numpy as np
 import tensorflow as tf
 from tensorflow import Tensor, Variable
-from tensorflow.keras.callbacks import Callback
+from tensorflow.python.keras.callbacks import Callback
```
Comment (guy1992l) [inline mlrun/frameworks/tf_keras/callbacks/logging_callback.py]:
```suggestion
if tf.__version__.split(".")[1] < 6:
    tensorflow.keras.callbacks import Callback
else:
    from keras.callbacks import Callback
```

File: mlrun/frameworks/tf_keras/mlrun_interface.py [hunk 2]
```diff
@@ -19,15 +19,15 @@
...
-from tensorflow.keras.callbacks import (
+from tensorflow.keras.optimizers import Optimizer
+from tensorflow.python.keras.callbacks import (
     BaseLogger,
     Callback,
     CSVLogger,
     ModelCheckpoint,
     ProgbarLogger,
     TensorBoard,
 )
```
Comment (guy1992l) [inline mlrun/frameworks/tf_keras/mlrun_interface.py]:
```suggestion
if tf.__version__.split(".")[1] < 6:
    from tensorflow.keras.callbacks import (
        BaseLogger,
        Callback,
        CSVLogger,
        ModelCheckpoint,
        ProgbarLogger,
        TensorBoard,
    )
else:
    from keras.callbacks import (
        BaseLogger,
        Callback,
        CSVLogger,
        ModelCheckpoint,
        ProgbarLogger,
        TensorBoard,
    )
```

**Review by guy1992l — CHANGES_REQUESTED** [review]
I've read more and since TensorFlow 2.6, Keras got separated and being released as a pip package (like its used to be before TensorFlow 1.12 I think). So, to get access to the `Callback`, please do as I suggested in the comments.

See here a tutorial from tensorflow: https://www.tensorflow.org/guide/keras/writing_your_own_callbacks?hl=en

IMO, we can say we only support TF 2.6 and on to skip the version checking... But lets keep it for now I guess (I'll ask Nir if we have anyone who use TF with lower versions).

**Review by guy1992l — APPROVED** [review]

**Review by assaf758 — APPROVED** [review]

**Pull request merged**